src/Angor/Client/NetworkConfiguration.cs

```diff
@@ -8,29 +8,75 @@ namespace Angor.Client;
 public class NetworkConfiguration : INetworkConfiguration
 {
     public static string AngorTestKey = "tpubD8JfN1evVWPoJmLgVg6Usq2HEW9tLqm6CyECAADnH5tyQosrL6NuhpL9X1cQCbSmndVrgLSGGdbRqLfUbE6cRqUbrHtDJgSyQEY2Uu7WwTL";
+    public static string AngorMainKey = "todo: add an Angor key here";
+
+    private Network currentNetwork;
+
+    public void SetNetwork(Network network)
+    {
+        currentNetwork = network;
+    }
 
     public Network GetNetwork()
     {
-        return new BitcoinSignet();
+        if (currentNetwork == null)
+        {
+            throw new ApplicationException("Network not set");
+        }
+
+        return currentNetwork;
     }
 
     public SettingsUrl GetIndexerUrl()
     {
-        return new SettingsUrl { Name = "", Url = "https://tbtc.indexer.angor.io/api" };
+        if (currentNetwork.NetworkType == NetworkType.Testnet)
+        {
+            return new SettingsUrl { Name = "", Url = "https://tbtc.indexer.angor.io/api" };
+        }
+
+        if (currentNetwork.NetworkType == NetworkType.Mainnet)
+        {
+            return new SettingsUrl { Name = "", Url = "https://btc.indexer.angor.io/api" };
+        }
+
+        throw new ApplicationException("Network not set");
     }
 
     public SettingsUrl GetExplorerUrl()
     {
+        if (currentNetwork.NetworkType == NetworkType.Testnet)
+        {
+            return new SettingsUrl { Name = "", Url = "https://explorer.angor.io/tbtc/explorer" };
+        }
+
+        if (currentNetwork.NetworkType == NetworkType.Mainnet)
+        {
+            return new SettingsUrl { Name = "", Url = "https://explorer.angor.io/btc/explorer" };
+        }
 
-        return new SettingsUrl { Name = "", Url = "https://explorer.angor.io/tbtc/explorer" };
+        throw new ApplicationException("Network not set");
     }
 
     public List<SettingsUrl> GetDefaultIndexerUrls()
     {
-        return new List<SettingsUrl>
+        if (currentNetwork.NetworkType == NetworkType.Testnet)
```

Comment:
Why not just replace the subdomain in the URL if the rest stays the same?

Reply:
it may not always be the case, it is just simpler this way.